### PR TITLE
[mkcal] Fix inter-process locking of sqlite storeage.

### PR DESCRIPTION
**Pull request by adenexter**
Constructing a QSystemSemaphore with the Create flag resets the
semaphore on construction releasing any lock held by another process
at that time, the Open flag is more correct but has issues with crashed
processes holding onto locks.  So use the inter-process mutex from
qmf/qtcontacts-sqlite which doesn't have these issues instead of
QSystemSemaphore.

### CODE REVIEW

File: src/semaphore_p.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE."
+ */
+
+#ifndef QTCONTACTSSQLITE_SEMAPHORE_P
+#define QTCONTACTSSQLITE_SEMAPHORE_P
+
```
Comment (VDVsx) [inline src/semaphore_p.h]: These ones should be changed to match this file.

Comment (adenexter) [inline src/semaphore_p.h]: Done.

**Comment by VDVsx** [comment]
LGTM, @pvuorela should also take a look since he's main devel in this area.

File: src/semaphore_p.cpp [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+void semaphoreError(const char *msg, const char *id, int error)
+{
+    kError() << QString::fromLatin1("%1 %2: %3 (%4)").arg(msg).arg(id).arg(::strerror(error)).arg(error).toUtf8().constData();
+}
+
+int semaphoreInit(const char *id, size_t count, const int *initialValues)
+{
+    int rv = -1;
+
+    key_t key = ::ftok(id, 0);
```
Comment (pvuorela) [inline src/semaphore_p.cpp]: Exists already in qtcontacts-sqlite, but notice man page for ftok() wanting non-zero least significant 8 bits on second parameter. @matthewvogt ?

**Comment by pvuorela** [comment]
One comment, on general LGTM.

File: src/semaphore_p.cpp (hunk 2 above)
Comment (matthewvogt) [inline src/semaphore_p.cpp]: Yeah, you're right, should probably change that.  Practically speaking, it's just used to disambiguate multiple users of the same token file, but we shouldn't contradict the man page! Any value will do.

Comment (adenexter) [inline src/semaphore_p.cpp]: Updated, the any value is 5.

**Comment by chriadam** [comment]
Merging this one.

**Pull request merged**